python/ray/ml/preprocessors/encoder.py

```diff
@@ -177,35 +185,59 @@ def _get_unique_value_indices(
     columns: List[str],
     drop_na_values: bool = False,
     key_format: str = "unique_values({0})",
+    limit: Optional[Dict[str, int]] = None,
 ) -> Dict[str, Dict[str, int]]:
     """If drop_na_values is True, will silently drop NA values."""
+    limit = limit or {}
+    for column in limit:
+        if column not in columns:
```

Comment:
nit:

if any(column not in columns for column in limit)

Reply:
Not easily converted to list comprehension as we need to print out related error msg.